frontend/js/src/brainzplayer/AppleMusicPlayer.tsx

```diff
@@ -0,0 +1,365 @@
+      this.appleMusicPlayer.pause();
+    } else {
+      console.log("Apple Music Player:", this.appleMusicPlayer);
+      console.log("#######################################");
+      this.appleMusicPlayer.authorize();
+      this.appleMusicPlayer.play();
+    }
+  };
+
+  stopAndClear = (): void => {
+    // eslint-disable-next-line react/no-unused-state
+    this.setState({ currentAppleMusicTrack: undefined });
+    if (this.appleMusicPlayer) {
+      this.appleMusicPlayer.pause();
+    }
+  };
+
+  handleAccountError = (): void => {
+    const errorMessage = (
+      <p>
+        In order to play music with AppleMusic, you will need a AppleMusic
+        Premium account linked to your ListenBrainz account.
+        <br />
+        Please try to{" "}
+        <a href="/profile/music-services/details/" target="_blank">
+          link for &quot;playing music&quot; feature
+        </a>{" "}
+        and refresh this page
+      </p>
+    );
+    const { onInvalidateDataSource } = this.props;
+    onInvalidateDataSource(this, errorMessage);
+  };
+
+  seekToPositionMs = (msTimecode: number): void => {
+    const timeCode = Math.floor(msTimecode / 1000);
+    this.appleMusicPlayer.seekToTime(timeCode);
+  };
+
+  disconnectAppleMusicPlayer = (): void => {
+    if (!this.appleMusicPlayer) {
+      return;
+    }
+    this.appleMusicPlayer.removeEventListener(
+      "playbackStateDidChange",
+      this._boundOnPlaybackStateChange
+    );
+    this.appleMusicPlayer.removeEventListener(
+      "playbackTimeDidChange",
+      this._boundOnPlaybackTimeChange
+    );
+    this.appleMusicPlayer.removeEventListener(
+      "playbackDurationDidChange",
+      this._boundOnPlaybackDurationChange
+    );
+    this.appleMusicPlayer.removeEventListener(
+      "nowPlayingItemDidChange",
+      this._boundOnNowPlayingItemChange
+    );
+    this.appleMusicPlayer = null;
+  };
+
+  connectAppleMusicPlayer = async (): Promise<void> => {
+    this.disconnectAppleMusicPlayer();
+
+    const musickit = window.MusicKit;
+    if (!musickit) {
+      setTimeout(this.connectAppleMusicPlayer.bind(this), 1000);
+      return;
+    }
+    await musickit.configure({
+      developerToken: "developer token here",
+      app: {
+        name: "ListenBrainz",
+        build: "latest",
+      },
+    });
+    this.appleMusicPlayer = musickit.getInstance();
+    await this.appleMusicPlayer.authorize();
+    this.appleMusicPlayer.addEventListener(
+      "playbackStateDidChange",
+      this._boundOnPlaybackStateChange
+    );
+    this.appleMusicPlayer.addEventListener(
+      "playbackTimeDidChange",
+      this._boundOnPlaybackTimeChange
+    );
+    this.appleMusicPlayer.addEventListener(
+      "playbackDurationDidChange",
+      this._boundOnPlaybackDurationChange
+    );
+    this.appleMusicPlayer.addEventListener(
+      "nowPlayingItemDidChange",
+      this._boundOnNowPlayingItemChange
+    );
+  };
+
+  onPlaybackStateChange = ({ state: currentState }: any) => {
```

Comment:
Rename this as `handlePlaybackStateChange` to make it consistent with other code, also it would be preferable to not use `any` here

Reply:
nowhere in apple music kit and in this file they are using handles in any name and hence I am not using it either. But if you still insist I change the name.